src/helpers/sceneless-project.ts

```diff
@@ -1534,6 +1549,48 @@ export const commands = (_project: ScenelessProject) => {
       })
     },
 
+    getRTMPNode(id: string) {
+      return content.children.find(
+        (x) =>
+          x.props.sourceProps?.id === id && x.props.sourceProps?.type === 'rtmp',
+      )
+    },
+
+    useRTMPNodes(cb) {
+      let nodeIds: string[] = []
+      const sendState = () => {
+        const nodes = content.children.filter((n) => n.props?.sourceProps?.type === 'rtmp')
+        nodeIds = nodes.map(n => n.id)
+        return cb(nodes)
+      }
+
+      sendState()
+
+      const nodeAddedListener = CoreContext.onInternal(
+        'NodeAdded',
+        (payload) => {
+          const node = _project.scene.get(payload.nodeId)
+          if (node?.props?.sourceProps?.type === 'rtmp') {
+            sendState()
+          }
+        }
+      )
+
+      const nodeRemovedListener = CoreContext.onInternal(
+        'NodeRemoved',
+        (payload) => {
+          if (nodeIds.indexOf(payload.nodeId) !== -1) {
+            sendState()
+          }
+        }
+      )
```

Comment:
Is there a possibility that you will update the node props during the RTMP lifecycle and also want to listen for 'NodeChanged' events?

Reply:
at the moment, no. maybe at a later time it will be changed